### PR TITLE
feat(cdp): improve testing interface

### PR DESCRIPTION
**Pull request by mariusandra**
## Changes

- Move "mock" and reload buttons into "..."
- Fix modifying project name when testing invocations
- Add a system to persist globals in localstorage

## How did you test this code?

Played around in the UI.

Persisting into localstorage isn't obvious though --> as a user I'd expect this to live in a database somewhere. Should we put this somewhere, or make the persistance class more clear in the UX?

### CODE REVIEW

File: frontend/src/scenes/pipeline/hogfunctions/HogFunctionTest.tsx [hunk 1]
```diff
@@ -102,39 +129,88 @@ export function HogFunctionTest(props: HogFunctionTestLogicProps): JSX.Element {
...
+                                                            fullWidth
+                                                            className="flex-1"
+                                                        >
+                                                            {name}
+                                                        </LemonButton>
+                                                        <LemonButton
+                                                            size="small"
+                                                            icon={<IconX />}
+                                                            onClick={() => deleteSavedGlobals(index)}
+                                                            tooltip="Delete this saved set of globals"
```
Comment (benjackwhite) [inline frontend/src/scenes/pipeline/hogfunctions/HogFunctionTest.tsx:185]: I think I'd opt for "save test data" or something as globals sounds a bit "internal"

File: frontend/src/scenes/pipeline/hogfunctions/hogFunctionTestLogic.tsx [hunk 2]
```diff
@@ -66,11 +69,23 @@ export const hogFunctionTestLogic = kea<hogFunctionTestLogicType>([
                 setTestResult: (_, { result }) => result,
             },
         ],
+
+        savedGlobals: [
+            [] as { name: string; globals: HogFunctionInvocationGlobals }[],
+            { persist: true, prefix: `${getCurrentTeamId()}__` },
```
Comment (benjackwhite) [inline frontend/src/scenes/pipeline/hogfunctions/hogFunctionTestLogic.tsx:75]: Personally I think this is fine. I'd say, lets add some tracking to this. If we find a lot of usage then we can consider a more robust persistence?

**Review by benjackwhite — APPROVED** [review]
I'd say this is more than good enough for an incremental change. If we add some tracking and see if anyone uses it I think thats the way to know whether to invest harder

File: frontend/src/scenes/pipeline/hogfunctions/hogFunctionTestLogic.tsx (hunk 2 above)
Comment (mariusandra) [reply frontend/src/scenes/pipeline/hogfunctions/hogFunctionTestLogic.tsx:75]: I added a bunch of `data-attr`-s all over the place and called these things "test data" everywhere. I guess good enough?

**Pull request merged**